### PR TITLE
Create OTA image from CMake

### PR DESCRIPTION
**Pull request by Damian-Nordic**

### CODE REVIEW

File: config/zephyr/Kconfig [hunk 1]
```diff
@@ -131,6 +170,20 @@ config APP_LINK_WITH_CHIP
 	  Add Connected Home over IP header files to the 'app' include path
 	  and link the 'app' with Connected Home over IP libraries.
 
+config CHIP_OTA_IMAGE_BUILD
+	bool "Generate OTA image"
+	help
+	  Enable building OTA (Over-the-air update) image. When this option is
+	  enabled, the resulting OTA image file will be created at
+	  build-dir/zephyr/matter-ota.bin.
```
Comment (kkasperczyk-no) [inline config/zephyr/Kconfig:178]: Correct me if I'm wrong, but isn't the matter-ota.bin name given by the nrfconnect script? So for other Zephyr platforms, like Telink it is not true.

Comment (Damian-Nordic) [reply config/zephyr/Kconfig:178]: Right.

Comment (Damian-Nordic) [reply config/zephyr/Kconfig:178]: Though... I'm not sure what path Telink will use. I wouldn't be surprised if it was the same. :)

File: config/zephyr/Kconfig [hunk 2]
```diff
@@ -34,6 +34,45 @@ menuconfig CHIP
...
+	default 0
+	range 0 4294967295
+	help
+	  Device software version is used by various CHIP components. For example,
+	  it is an attribute of the Basic Information cluster and it is included in
+	  OTA (Over-the-air update) image header. Note that due to the rollback
+	  protection a device will only accept software updates with versions
+	  greater than its current one.
+
+config CHIP_DEVICE_SOFTWARE_VERSION_STRING
```
Comment (kkasperczyk-no) [inline config/zephyr/Kconfig:66]: According to spec the software version string attribute in BasicInformationCluster maximum length is 64 bytes of UTF-8 characters. I'm not sure if Kconfig can do such check, but maybe we could at least add this information to help?

File: config/zephyr/Kconfig (hunk 1 above)
Comment (kkasperczyk-no) [reply config/zephyr/Kconfig:178]: Maybe another kconfig specifying this path would help?

Comment (Damian-Nordic) [reply config/zephyr/Kconfig:178]: I'll remove the comment. Not sure if there's much value in allowing to define this path. No other Zephyr paths are configurable I think.

File: config/zephyr/Kconfig (hunk 2 above)
Comment (Damian-Nordic) [reply config/zephyr/Kconfig:66]: Ok. I'll add.

File: config/zephyr/Kconfig (hunk 1 above)
Comment (Damian-Nordic) [reply config/zephyr/Kconfig]: Done

File: config/zephyr/Kconfig (hunk 2 above)
Comment (Damian-Nordic) [reply config/zephyr/Kconfig:66]: Done

**Review by kkasperczyk-no — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]
Looks great! :+1:

**Pull request closed without merging**